src/scripts/main.js

```diff
@@ -1,11 +1,26 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    const handleEvent = (e) => {
+      element.removeEventListener(eventName, handleEvent);
+
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    };
+
+    element.addEventListener(eventName, handleEvent);
+  });
```

Comment:
You should consider error handling in your Promise. Right now, if the event does not occur, the Promise will be pending indefinitely. You could add a timeout to reject the Promise after a certain period.